### PR TITLE
feat(walk): add walk command for querying OpenAPI specs

### PR DESCRIPTION
**Pull request by erraggy**
## Summary

- Add `oastools walk` with 6 subcommands for exploring OpenAPI documents: `operations`, `schemas`, `parameters`, `responses`, `security`, and `paths`
- Architecture follows a Collect-Filter-Render pattern with indexed collector APIs in the `walker` package
- Includes an extension filter DSL with AND/OR logic, negation, value matching, path glob matching, and status code wildcards

## Subcommands

| Subcommand | Filters |
|---|---|
| `operations` | `--method`, `--path`, `--tag`, `--deprecated`, `--operationId` |
| `schemas` | `--name`, `--type`, `--component`, `--inline` |
| `parameters` | `--name`, `--in`, `--path`, `--method` |
| `responses` | `--status` (supports `2xx` wildcards), `--path`, `--method` |
| `security` | `--name`, `--type` |
| `paths` | `--path` (supports `*` glob) |

All subcommands share `--format` (text/json/yaml), `-q`/`--quiet`, `--detail`, `--extension`, and `--resolve-refs`.

## Collector APIs

New `walker` package collector functions with indexed lookups:

- `CollectOperations` → `ByPath`, `ByMethod`, `ByTag`
- `CollectSchemas` → `ByPath`, `ByName`, `Components`, `Inline`
- `CollectParameters` → `ByName`, `ByLocation`, `ByPath`
- `CollectResponses` → `ByStatusCode`, `ByPath`
- `CollectSecuritySchemes` → `ByName`

## Extension filter DSL

```
x-internal=true          # value match
!x-deprecated            # negated existence
x-foo!=bar               # negated value
x-a+x-b=1               # AND (both must match)
x-foo,x-bar              # OR (either matches)
x-a+x-b=1,!x-c          # mixed AND+OR
```

## Examples

```bash
oastools walk operations api.yaml
oastools walk operations --method get --path /pets --detail api.yaml
oastools walk schemas --name Pet --detail api.yaml
oastools walk operations --extension x-audited-by api.yaml
oastools walk responses --status '4xx' -q --format json api.yaml | jq
```

## Test plan

- [x] Unit tests for all 6 subcommands with filter combinations
- [x] Extension filter parser/matcher tests including edge cases (double negation, bare `!`)
- [x] `FormatExtensions` deterministic output tests
- [x] `matchPath` glob and `matchStatusCode` wildcard tests
- [x] Collector tests with `ByName`, `ByLocation`, `ByStatusCode`, `ByPath` assertions
- [x] Empty document and nil/error handling tests
- [x] `make check` passes (7,847 tests, lint clean)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a new "walk" command with subcommands to explore operations, schemas, parameters, responses, security schemes, and paths; supports rich filtering and output modes (detail, summary, quiet) in JSON/YAML/Text.
  * Extension-filter syntax for AND/OR matching and deterministic extension display.
  * Rendering utilities for summary tables and detailed output; collectors for parameters, responses, and security schemes.

* **Documentation**
  * Added design and implementation plans for the walk command.

* **Tests**
  * Extensive unit and integration tests covering collectors, filters, rendering, subcommands, and CLI behavior.

### CODE REVIEW

File: cmd/oastools/commands/walk_render_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+	err := RenderDetail(&buf, node, FormatJSON, false)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	output := buf.String()
+	if !strings.Contains(output, `"summary"`) {
+		t.Error("expected summary in JSON output")
+	}
+}
```
[the remark this answers is not in the document]
Comment (erraggy) [reply cmd/oastools/commands/walk_render_test.go]: Won't fix — same reasoning as the parameters test comment. No tests in `cmd/oastools/commands/` use testify. Converting all command tests to testify is a good idea but belongs in a dedicated follow-up, not mixed into a feature PR.

File: cmd/oastools/commands/walk_security.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		if schemeType != "" && !strings.EqualFold(info.SecurityScheme.Type, schemeType) {
+			continue
+		}
+		if extFilter != nil && !extFilter.Match(info.SecurityScheme.Extra) {
+			continue
+		}
+		matched = append(matched, info)
+	}
+	return matched, nil
+}
```
[the remark this answers is not in the document]
Comment (erraggy) [reply cmd/oastools/commands/walk_security.go]: Fixed — added `if info == nil || info.SecurityScheme == nil { continue }` guard at the top of the `filterSecuritySchemes` loop. The summary rendering loop is safe because it only iterates over entries that passed the filter.

File: cmd/oastools/commands/walk.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+		return handleWalkResponses(subArgs)
+	case "security":
+		return handleWalkSecurity(subArgs)
+	case "paths":
+		return handleWalkPaths(subArgs)
+	default:
+		printWalkUsage()
+		return fmt.Errorf("unknown walk subcommand: %s", subcommand)
+	}
+}
```
[the remark this answers is not in the document]
Comment (erraggy) [reply cmd/oastools/commands/walk.go]: Won't fix — the switch is explicit and readable with 6 entries. The usage text includes descriptions ("List or inspect operations") that wouldn't map cleanly to a `map[string]func`. With a small, fixed set of subcommands, the explicitness of a switch is preferable to the indirection of a map lookup. If we add significantly more subcommands, this refactor would make sense then.

File: cmd/oastools/commands/walk.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+Examples:
+  oastools walk operations api.yaml
+  oastools walk operations --method get --path /pets --detail api.yaml
+  oastools walk schemas --name Pet --detail api.yaml
+  oastools walk operations --extension x-audited-by api.yaml
+  oastools walk responses --status '4xx' -q --format json api.yaml | jq
+
+Run 'oastools walk <subcommand> --help' for subcommand-specific flags.
+`)
+}
```
[the remark this answers is not in the document]
Comment (erraggy) [reply cmd/oastools/commands/walk.go]: Fixed — updated the example to include `--detail` so it accurately reflects that `--format json` applies to detail output: `oastools walk responses --status '4xx' -q --detail --format json api.yaml | jq`

File: docs/plans/2026-02-12-walk-command-design.md [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+# Parameters of a specific operation
+oastools walk parameters --path /pets --method get api.yaml
+
+# 4xx responses, piped to jq
+oastools walk responses --status '4xx' -q --format json api.yaml | jq '.[]'
+
+# Works identically on OAS 2.0 and 3.x
+oastools walk schemas --name Pet --detail swagger.yaml
+oastools walk schemas --name Pet --detail openapi.yaml
+```
```
[the remark this answers is not in the document]
Comment (erraggy) [reply docs/plans/2026-02-12-walk-command-design.md]: plans are transient for completing efforts. No need to polish them now.
